src/concurrency/init_once.rs

```diff
@@ -201,4 +185,19 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
 
         Ok(())
     }
+
+    /// Synchronize with the previous completion or failure of an InitOnce.
+    /// This is required to prevent data races.
+    #[inline]
+    fn init_once_acquire(&mut self, id: InitOnceId) {
```

Comment:
Please keep the name consistent with the locks -- `validate_init_once_acquie` maybe? Also this should probably be on a private extension trait.

Reply:
Okay never mind this part, I only later realized how this function is being used.